src/gui/processing/models/qgsmodelarrowitem.cpp

```diff
@@ -83,9 +87,38 @@ void QgsModelArrowItem::paint( QPainter *painter, const QStyleOptionGraphicsItem
   painter->setBrush( color );
   painter->setRenderHint( QPainter::Antialiasing );
 
-  for ( const QPointF &point : qgis::as_const( mNodePoints ) )
+  if ( mStartMarker == Marker::Circle )
```

Comment:
use a switch statement here instead (if you omit the "default:" branch, then the compiler will raise warnings if someone ever adds another marker type, giving them a direct pointer to all the code paths which need to be updated as a result!)